### DISCUSSION TITLE
Math formula display optimization needed for specific question in OpenWebUI

### DISCUSSION THREAD

**Discussion by sandycs-protoss** [open-webui/open-webui#4008] [General]
Description

When entering the question "A five-digit number with its last three digits as 999. If this five-digit number is divisible by 23, what is the smallest such number?" in OpenWebUI, the resulting answer contains mathematical formulas that are not displayed in an optimized format. This affects the readability and clarity of the provided solution.

Steps to Reproduce
Open OpenWebUI.
Enter the question: "A five-digit number with its last three digits as 999. If this five-digit number is divisible by 23, what is the smallest such number?"
Observe the output.
Expected Result
The mathematical formulas should be displayed in an optimized manner for easier understanding and readability.
Actual Result
The mathematical formulas are not displayed optimally, making the content difficult to read and understand.
OpenWebUI Version: v0.3.6

Additional Information
Please refer to the screenshot below to understand the specific issue:
Problem Screenshot

![image](https://github.com/user-attachments/assets/c4be0a70-8c60-4207-97d7-f570cc38c2f3)

Possible Solution
Consider adding support for LaTeX to allow mathematical expressions to be displayed more clearly and standardized.
Thank you to the development team for your hard work. I hope this issue can be resolved soon.

**Comment by hruzgar** [comment]
Just wanted to say that it would be also very useful for me as I can't use this ui for math right now

**Comment by Peter-De-Ath** [comment]
I believe this has been resolved in the latest version of Open WebUI
https://github.com/open-webui/open-webui/commit/41fecb800d0f66b6aaa6edbf1af1563229f78130


![image](https://github.com/user-attachments/assets/2531037a-05ed-401a-a832-9727366c4231)

**Comment by cduk** [comment]
How can we get math rendered more nicely. Currently it looks like this:

![image](https://github.com/user-attachments/assets/21c226cc-5941-45ad-9789-3faee1274792)

compare to OpenAI chat interface:

![image](https://github.com/user-attachments/assets/d116721a-dfe3-43d0-a378-1372b7b638c1)

Which is far easier to read.
